extensions/amp-story/1.0/amp-story-interactive-confetti.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {htmlFor} from '../../../src/static-template';
+import {setStyles} from '../../../src/style';
+
+/**
+ * Generates the template for the particle wrapper.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildWrapperTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div
+      class="i-amphtml-story-interactive-particle-wrapper"
+      aria-hidden="true"
+    ></div>
+  `;
+};
+
+/**
+ * Generates a template for a particle.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildParticleTemplate = (element) => {
+  const html = htmlFor(element);
+  return html` <div class="i-amphtml-story-interactive-particle"></div> `;
+};
+
+/**
+ * Creates a radial burst of emoji.
+ * The burst begins from the center of the parent element and
+ * animates to the edges of the parent element.
+ * Nodes are removed from the dom at the end of the animation.
+ *
+ * @param {!Element} element
+ * @param {string} confettiEmoji
+ * @return {void}
+ */
+export function emojiBurst(element, confettiEmoji) {
```

Comment:
Nit: Would be cool to change the name of the parameter to rootEl / rootElement to signify you pass in the root element (this.rootEl_ inside the shadow dom) instead of the this.element (the amp-sotry-interactive-x) element.